src/NuGet.Services.AzureSearch/SearchService/SearchResponseBuilder.cs

```diff
@@ -384,6 +385,22 @@ private V3SearchPackage ToV3SearchPackage(SearchDocument.Full result, string reg
             };
         }
 
+        private List<V3SearchPackageType> GetV3SearchPackageTypes(SearchDocument.UpdateLatest document)
+        {
+            if (document.PackageTypes == null || document.PackageTypes.Length == 0)
+            {
+                return null;
```

Comment:
This case is a pretty good indication that we are running new search service against an old index. Do we expect this to be a normal case at any point? or should we log something so we are aware? I don't expect us to ever spend any extended time in this state, but just in case (also sanity check once we go full new index).

Reply:
I chose not to log in this case because it would be very verbose -- essentially once per search entry.

Reply:
> Do we expect this to be a normal case at any point?

Normal from our roll-out plan perspective - no. We should rebuild first, then deploy these bits. But have having catastrophic failure in this case could be painful during the rollout (e.g. deploying a build to DEV environment prior to the rebuild).